pkg/shared/util.go

```diff
@@ -87,3 +103,35 @@ func setTLSOptions(tlsName **C.char, tlsConfig *C.as_config_tls, tls *model.TLS)
 		setCString(&tlsConfig.certfile, tls.Certfile)
 	}
 }
+
+//nolint:gocritic
+func configureEncryption(encryptMode *C.encryption_opt, pKey **C.encryption_key_t,
+	policy *model.EncryptionPolicy) {
+	if policy != nil {
+		switch policy.Mode {
+		case model.EncryptNone:
+			*encryptMode = C.IO_PROXY_ENCRYPT_NONE
+		case model.EncryptAES128:
+			*encryptMode = C.IO_PROXY_ENCRYPT_AES128
+		case model.EncryptAES256:
+			*encryptMode = C.IO_PROXY_ENCRYPT_AES256
+		}
+		if policy.KeyFile != nil {
+			*pKey = C.enc_key_malloc()
```

Comment:
do we need to free this memory?

Reply:
It is checked for null and freed in the config destroy functions.